src/bin/pg_autoctl/parsing.c

```diff
@@ -243,6 +248,66 @@ parse_controldata(PostgresControlData *pgControlData,
 }
 
 
+#define streq(x, y) ((x != NULL) && (y != NULL) && (strcmp(x, y) == 0))
+
+/*
+ * parse_controldata_field_dbstate matches pg_controldata output for Database
+ * cluster state and fills in the value string as an enum value.
+ */
+static bool
+parse_controldata_field_dbstate(const char *controlDataString, DBState *state)
+{
+	char regex[BUFSIZE] = { 0 };
+
+	sformat(regex, BUFSIZE, "Database cluster state: *(.*)$");
+
+	char *match = regexp_first_match(controlDataString, regex);
+
+	if (match == NULL)
+	{
+		return false;
+	}
+
+	if (streq(match, "starting up"))
```

Comment:
Does this still work if you use a non English localization for PG? i.e. if you use French, does it still store the english strings in there? I mainly worry about this because of your other localization related stuff.

Reply:
Yes it does, we force LANGUAGE=C when running pg_controldata.